### PR TITLE
binned and parametrized Z non-closure nuisances

### PR DESCRIPTION
**Pull request by tyjyang**

### CODE REVIEW

File: wremnants/include/muon_calibration.h [hunk 1]
```diff
@@ -677,26 +704,126 @@ class JpsiCorrectionsUncHelper {
...
+    template<typename... Xs>
+    const tensor_t &get_tensor(const Xs&... xs) {
+        return get_tensor_impl(std::index_sequence_for<Xs...>{}, xs...);
+    }
+
+    out_tensor_t operator() (
+        double genQop, float genEta,
+        double recoQop, float recoEta, int recoCharge, double recoPt,  
+        const RVec<float> &cov, double nominal_weight = 1.0,
+        std::vector<std::string> vars =  {"A", "M"}
```
Comment (kdlong) [inline wremnants/include/muon_calibration.h:734]: Better to do these comparisons with enums than with strings. Or you can  just have three booleans: doA, doM, doE.

Comment (tyjyang) [reply wremnants/include/muon_calibration.h]: done [here](https://github.com/WMass/WRemnants/pull/200/commits/5129189730420d818d1254c283a5d5b65aae2ed2)

**Comment by tyjyang** [comment]
added option to produce the non-closure nuisances decorrelated in phase space

**Comment by kdlong** [comment]
Ok for me. @bendavid ?

@tyjyang  Do you have the impacts for the full stat analysis?

**Comment by tyjyang** [comment]
> Ok for me. @bendavid ?
> 
> @tyjyang Do you have the impacts for the full stat analysis?

Here is the impacts with the correlated parametrized nuisance:
```
Impact of all systematics (in MeV)
   resumTNP: 0.17
   muon_eff_syst_reco: 0.97
   muon_eff_stat_iso_effData: 2.32
   muon_eff_stat_trigger: 1.32
   muon_eff_stat_tracking: 0.47
   muonScale_nonClosure_parametrized: 1.9
   muon_eff_syst_iso: 0.67
   muon_eff_syst_tracking: 0.46
   CMS_recoil: 0.48
   muon_eff_stat_idip: 0.32
   QCDscalePtChargeMiNNLO: 1.68
   muon_eff_stat_iso_effMC: 1.5
   muonScale: 1.18
   resumTransition: 0.41
   muon_eff_syst_trigger: 0.79
   muon_eff_stat_reco: 0.94
   luminosity: 0.38
   muonPrefire: 0.49
   ecalPrefire: 0.15
   muon_eff_stat: 3.45
   resumNonpert: 0.51
   muon_eff_syst: 1.41
   pdfMSHT20: 5.26
   muon_eff_syst_idip: 0.05
   stat: 0.0
   binByBinStat: 6.36
   Total: 9.04
```

**Comment by bendavid** [comment]
So I think what we want rather as default for now  is the uncorrelated parameterized case (with A and M split), and also the standard stat uncertainties with the usual inflation factor.  (even all included in a single impact group at this point)

**Comment by tyjyang** [comment]
@bendavid @kdlong

- de-correlated, parametrized non-closure nuisances with A, M separated is now the default
- implemented default flat inflation factor bloating data-only stats. unc. to data+MC stats. unc. for the jpsi massfit
- non-closure and jpsi massfit stats. unc. nuisances are now in one impact group by default

If there's no remaining issues, I propose that we merge this PR.

**Pull request merged**